pkg/arvo/sys/vane/gall.hoon

```diff
@@ -2,1457 +2,2882 @@
-    ::::::::::::::::::::::::::::::::::::::::::::::::::::::
+::
+::  (state machine)
+::
+|%
+::
+::  +gall-payload:  gall payload.
+::
+++  gall-payload  +
+::
+::  +mo: Arvo-level move handling.
+::
+::  An outer core responsible for routing moves to and from Arvo; it calls
+::  an inner core, +ap, to route internal moves to and from agents.
+::
 ++  mo
   ~%  %gall-mo  +>  ~
-  =*  mas  mast.all
-  |_  $:  hen=duct
-          moz=(list move)
-      ==
-  ++  mo-abed                                           ::  initialize
+  ::
+  |_
+    $:
+      hen=duct
+      moves=(list move)
+    ==
+  ::
+  ++  mo-state  .
+  ::
+  ::  +mo-abed: initialise state with the provided duct.
+  ::
+  ++  mo-abed
     |=  =duct
-    ^+  +>
-    +>(hen duct)
-  ::
-  ++  mo-abet                                           ::  resolve to
-    ^+  [*(list move) +>+]
-    :_  +>+
-    %-  flop
-    %+  turn  moz
-    |=  a/move
-    ?.  ?=($pass -.q.a)  a
-    [p.a %pass p.q.a q.q.a]
-  ::
-  ++  mo-conf                                           ::  configure
-    |=  {dap/dude lum/culm}
-    (mo-boot dap p.p.lum q.p.lum da+now)
-  ::
-  ++  mo-pass                                           ::  standard pass
-    |=  {pax/path noh/note-arvo}
-    %_(+> moz :_(moz [hen %pass pax noh]))
+    ^+  mo-state
+    ::
+    mo-state(hen duct)
+  ::
+  ::  +mo-abet: resolve moves.
+  ::
+  ++  mo-abet
+    ^-  [(list move) _gall-payload]
+    ::
+    =/  resolved  (flop moves)
+    [resolved gall-payload]
+  ::
+  ::  +mo-boot: ask %ford to build us a core for the specified agent.
+  ::
+  ++  mo-boot
+    |=  [=term =ship =desk]
+    ^+  mo-state
+    ::
+    =/  =case  [%da now]
+    ::
+    =/  =path
+      =/  ship  (scot %p ship)
+      =/  case  (scot case)
+      /sys/core/[term]/[ship]/[desk]/[case]
+    ::
+    =/  =note-arvo
+      =/  =schematic:ford  [%core [ship desk] /hoon/[term]/app]
+      [%f %build live=%.y schematic]
+    ::
+    =/  pass  [path note-arvo]
+    (mo-pass pass)
+  ::
+  ::  +mo-pass: prepend a standard %pass to the current list of moves.
+  ::
+  ++  mo-pass
+    |=  pass=(pair path note-arvo)
+    ^+  mo-state
+    ::
+    =/  =move  [hen [%pass pass]]
+    mo-state(moves [move moves])
+  ::
+  ::  +mo-give: prepend a standard %give to the current list of moves.
   ::
   ++  mo-give
-    |=  git/gift:able
-    %_(+> moz :_(moz [hen %give git]))
+    |=  =gift:able
+    ^+  mo-state
+    ::
+    =/  =move  [hen [%give gift]]
+    mo-state(moves [move moves])
   ::
-  ++  mo-okay                                           ::  valid agent core
-    ~/  %mo-okay
-    |=  vax/vase
+  ::  +mo-contains-valid-bowl: check that a vase contains a valid bowl.
+  ::
+  ++  mo-contains-valid-bowl
+    ~/  %mo-contains-valid-bowl
+    |=  =vase
     ^-  ?
-    =+  bol=(slew 12 vax)
-    ?~  bol  |
-    (~(nest ut p.u.bol) %| -:!>(*bowl))
-  ::  +mo-receive-core: receives an app core built by ford-turbo
+    ::
+    =/  inferred  -:!>(*bowl)
```

Comment:
I found this variable name misleading, and it's defined a bit far away from its use.  Not egregious, but I think this would actually be clearer stuck into the +nest call, with no name.  You could call it `bowl-type`, but `-:!>` is immediately recognizable to a Hoon programmer as the type spear, so I don't think it's worth naming.

Conversely, I'd try to name `maybe-vase` and `p.u.maybe-vase` a bit more descriptively.  Something like `app-sample-head` or something to convey why we're doing this.